kernel/kernel-api/src/main/java/io/delta/kernel/utils/DataFileStatus.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.utils;
+
+import java.util.Optional;
+
+/**
+ * Extends {@link FileStatus} to include additional details such as column level statistics
+ * of the data file in the Delta Lake table.
+ */
+public class DataFileStatus extends FileStatus {
```

Comment:
Should we just change the `FileStatus` to include the `DataFileStatistics`? This class doesn't seem to be adding a lot of value.

Reply:
Surely there are cases of `FileStatus` where statistics just are not useful or applicable? This class seems very useful to me. Please elaborate?